src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -163,7 +163,7 @@ public String actualPathForPublicPath(String publicPath) {
             }
             final var exerciseId = Long.parseLong(shouldBeExerciseId);
             final var submissionId = Long.parseLong(shouldBeSubmissionId);
-            return FileUploadSubmission.buildFilePath(exerciseId, submissionId);
+            return FileUploadSubmission.buildFilePath(exerciseId, submissionId) + filename;
```

Comment:
how can this work for old files?

Reply:
For file upload submission we only use this method when deleting file associated with the submission. Before it was removing this file with the directory it was located in, now it will remove only the file itself. So this will not affect old files